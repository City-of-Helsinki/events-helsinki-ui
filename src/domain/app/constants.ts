export enum ROUTES {
  ABOUT = '/about',
  COLLECTION = '/collection/:id',
  COLLECTIONS = '/collections',
  EVENTS = '/events',
  EVENT = '/event/:id',
  HOME = '/home',
  HOME_PREVIEW = '/home/:id',
}
