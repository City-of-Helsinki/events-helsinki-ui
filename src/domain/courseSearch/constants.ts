export const DEFAULT_SEARCH_FILTERS = {
  categories: [],
  dateTypes: [],
  divisions: [],
  end: null,
  isFree: false,
  onlyOngoingCourses: false,
  places: [],
  start: null,
  text: [],
};
