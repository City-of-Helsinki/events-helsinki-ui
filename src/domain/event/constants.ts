export const mockEventData = {
  eventDetails: {
    description: {
      en: "description en",
      fi: "description fi",
      sv: "description sv"
    },
    endTime: null,
    externalLinks: [
      {
        link: "https://www.facebook.com/events/12345687/",
        name: "extlink_facebook"
      }
    ],
    id: "123",
    images: [
      {
        id: "60578",
        name: "piha-2.jpg",
        url: "https://api.hel.fi/linkedevents/media/images/piha-2_yYlCdPB.jpg"
      }
    ],
    inLanguage: [
      {
        id: "fi",
        name: {
          en: "Finnish",
          fi: "suomi",
          sv: "finska"
        },
        translationAvailable: true
      }
    ],
    infoUrl: {
      en:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      fi:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080",
      sv:
        "https://www.lippu.fi/tickets.html?affiliate=adv&doc=erdetaila&fun=erdetail&erid=2640080"
    },
    keywords: [],
    location: {
      addressLocality: {
        en: "locality en",
        fi: "locality fi",
        sv: "locality sv"
      },
      divisions: [],
      email: "location@info.com",
      infoUrl: {
        en: "http://www.infourl.com",
        fi: "http://www.infourl.com",
        sv: "http://www.infourl.com"
      },
      name: {
        en: "name en",
        fi: "name fi",
        sv: "name sv"
      },
      position: {
        coordinates: [25.01497, 60.250507],
        type: "Point"
      },
      postalCode: "01234",
      streetAddress: {
        en: "streetAddress en",
        fi: "streetAddress fi",
        sv: "streetAddress sv"
      },
      telephone: {
        en: "044 123 4567",
        fi: "044 123 4567",
        sv: "044 123 4567"
      }
    },
    name: {
      en: "name en",
      fi: "name fi",
      sv: "name sv"
    },
    offers: [
      {
        description: null,
        infoUrl: null,
        isFree: true,
        price: null
      }
    ],
    provider: null,
    publisher: "provider:123",
    shortDescription: {
      en: "short description en",
      fi: "short description fi",
      sv: "short description sv"
    },
    startTime: "2019-12-13T08:00:00Z"
  }
};

// Amount of the similar events
export const SIMILAR_EVENTS_AMOUNT = 8;

export const EVENT_KEYWORD_BLACK_LIST = [
  "helmet:10592", // Yleinen
  "helmet:10689", // Helsinki
  "helmet:11916", // Infonäytöt (Helsinki)
  "kulke:53", // Suomi
  "kulke:54", // Ruotsi
  "kulke:55", // Englanti
  "kulke:616", // Näkyy TV-sovelluksessa
  "kulke:671", // Normaali
  "kulke:673" // Suomi (TV)
];
