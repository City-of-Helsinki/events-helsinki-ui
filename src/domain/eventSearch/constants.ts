// Page size of the event list
export const PAGE_SIZE = 10;

export enum EVENT_SORT_OPTIONS {
  DURATION = 'duration',
  DURATION_DESC = '-duration',
  END_TIME = 'end_time',
  END_TIME_DESC = '-end_time',
  LAST_MODIFIED_TIME = 'last_modified_time',
  LAST_MODIFIED_TIME_DESC = '-last_modified_time',
  START_TIME = 'start_time',
  START_TIME_DESC = '-start_time',
}

export enum EVENT_SEARCH_FILTERS {
  CATEGORIES = 'categories',
  DATE_TYPES = 'dateTypes',
  DIVISIONS = 'divisions',
  END = 'end',
  IS_FREE = 'isFree',
  KEYWORD_NOT = 'keywordNot',
  KEYWORDS = 'keywords',
  ONLY_CHILDREN_EVENTS = 'onlyChildrenEvents',
  PAGE = 'page',
  PLACES = 'places',
  PUBLISHER = 'publisher',
  START = 'start',
  TEXT = 'text',
}

export const CULTURE_KEYWORDS = [
  'kulke:33', // Teatteri
  'kulke:51', // Sirkus
  'kulke:205', // Elokuva ja media
  'kulke:351', // Teatteri ja sirkus
  'matko:teatteri', // teatteri
  'yso:p360', // cultural events
  'yso:p1235', // films
  'yso:p1278', // dance (performing arts)
  'yso:p1808', // music
  'yso:p2625', // in Finnish teatteritaide, "theatre arts"
  'yso:p2739', // fine arts
  'yso:p2850', // performing arts
  'yso:p2851', // art
  'yso:p4934', // museums
  'yso:p5121', // exhibitions
  'yso:p6889', // art exhibitions
  'yso:p7969', // literary art
  'yso:p8113', // literature
  'yso:p8144', // art museums
  'yso:p9592', // modern art
  'yso:p9593', // contemporary art
  'yso:p10105', // contemporary dance
  'yso:p16327', // cinema (art forms)
];

export const INFLUENCE_KEYWORDS = [
  'yso:p1657', // Vaikuttaminen
  'yso:p10727', // Osallistuminen
];

export const MUSEUM_KEYWORDS = [
  'matko:museo', // Museo
  'yso:p4934', // Museot
];
