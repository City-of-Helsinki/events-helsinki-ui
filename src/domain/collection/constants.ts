import { mockEventData } from "../event/constants";

export const mockCollection = {
  collectionDetails: {
    curatedEvents: [
      `http://localhost:3000/fi/event/${mockEventData.eventDetails.id}`
    ],
    curatedEventsTitle: {
      en: "At least visit these",
      fi: "Käy ainakin näissä",
      sv: "Besök åtminstone dessa"
    },
    description: {
      en:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      fi:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus.",
      sv:
        // eslint-disable-next-line max-len
        "Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Nam quam nunc, blandit vel, luctus pulvinar, hendrerit id, lorem. Maecenas nec odio et ante tincidunt tempus. Donec vitae sapien ut libero venenatis faucibus."
    },
    eventListQuery:
      "http://localhost:3000/en/events?categories=music&districts=kaupunginosa%3Aetu-t%C3%B6%C3%B6l%C3%B6",
    eventListTitle: {
      en: "All the best events of the fall",
      fi: "Kaikki syksyn parhaat tapahtumat",
      sv: "Höstens bästa händelser"
    },
    id: "1",
    linkText: {
      en: "Read more on the project website",
      fi: "Lue lisää hankkeen omilta sivuilta",
      sv: "Läs mer på projektets webbplats"
    },
    linkUrl: {
      en: "http://www.google.com",
      fi: "http://www.google.com",
      sv: "http://www.google.com"
    },
    similarCollectionsTitle: {
      en: "Check also these",
      fi: "Katso myös nämä",
      sv: "Se dessa också"
    },
    socialMediaDescription: {
      en: "We put together the best foliage for the fall",
      fi: "Kokosimme parhaat tärpit syksylle",
      sv: "Vi sätter ihop det bästa bladverket för hösten"
    },
    subtitles: {
      en: "Subtitles en",
      fi: "Subtitles fi",
      sv: "Subtitles sv"
    },
    title: {
      en: "The most bustling tricks of the fall",
      fi: "Syksyn riehakkaimmat riennot",
      sv: "Höstens mest livliga knep"
    }
  }
};

// Amount of the similar collections
export const SIMILAR_COLLECTIONS_AMOUNT = 8;
