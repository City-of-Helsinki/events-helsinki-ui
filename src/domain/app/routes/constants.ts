import { MAPPED_PLACES } from '../../eventSearch/constants';

// Remember also update static urls to updateSitemap
export type ROUTE_TYPES = {
  ABOUT: '/about';
  ACCESSIBILITY: '/accessibility';
  COLLECTION: '/collection/:slug';
  COLLECTIONS: '/collections';
  EVENTS: '/events';
  EVENT: '/event/:id';
  COURSES: '/courses';
  COURSE: '/course/:id';
  HOME: '/home';
  HOME_PREVIEW: '/home/:id';
  EVENT_PLACE: string;
};

export const ROUTES: ROUTE_TYPES = {
  ABOUT: '/about',
  ACCESSIBILITY: '/accessibility',
  COLLECTION: '/collection/:slug',
  COLLECTIONS: '/collections',
  EVENTS: '/events',
  EVENT: '/event/:id',
  COURSES: '/courses',
  COURSE: '/course/:id',
  HOME: '/home',
  HOME_PREVIEW: '/home/:id',
  EVENT_PLACE: `/:place(${Object.keys(MAPPED_PLACES).join('|')})`,
};
