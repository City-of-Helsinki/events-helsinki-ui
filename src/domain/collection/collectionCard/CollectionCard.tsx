import classNames from 'classnames';
import { IconArrowRight } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { useHistory, useLocation } from 'react-router';
import { Link } from 'react-router-dom';

import IconButton from '../../../common/components/iconButton/IconButton';
import { CollectionFieldsFragment } from '../../../generated/graphql';
import useLocale from '../../../hooks/useLocale';
import { ROUTES } from '../../app/routes/constants';
import { getCollectionFields } from '../CollectionUtils';
import styles from './collectionCard.module.scss';

export type CollectionCardSize = 'sm' | 'md' | 'lg';

export interface Props {
  collection: CollectionFieldsFragment;
  size: CollectionCardSize;
}

const CollectionCard: React.FC<Props> = ({ collection, size }) => {
  const history = useHistory();
  const { search } = useLocation();
  const { t } = useTranslation();
  const locale = useLocale();
  const button = React.useRef<HTMLDivElement>(null);
  const {
    count,
    description,
    heroBackgroundImage: backgroundImage,
    slug,
    title,
  } = getCollectionFields(collection, locale);

  const collectionUrl = `/${locale}${ROUTES.COLLECTION.replace(
    ':slug',
    slug
  )}${search}`;

  const handleLinkClick = (ev: React.MouseEvent<HTMLAnchorElement>) => {
    const target = ev.target;
    if (button.current?.contains(target as Node)) {
      ev.preventDefault();
    }
  };

  const moveToCollectionPage = () => {
    history.push(collectionUrl);
  };

  const showDescription = size === 'lg';

  return (
    <Link
      aria-label={t('collection.collectionCard.ariaLabelLink', {
        title,
      })}
      className={classNames(
        styles.collectionCardWrapper,
        styles[`${size}Size`]
      )}
      onClick={handleLinkClick}
      to={collectionUrl}
    >
      <div
        aria-hidden={true}
        className={styles.imageWrapper}
        style={{ backgroundImage: `url(${backgroundImage})` }}
      ></div>
      <div className={styles.textWrapper}>
        <div className={styles.countWrapper}>
          <div className={styles.count}>
            {t('collection.collectionCard.count', { count })}
          </div>
        </div>

        <div className={styles.titleWrapper}>
          <span className={styles.title}>{title}</span>
          {showDescription && !!description && (
            <div className={styles.description}>{description}</div>
          )}
        </div>

        <div className={styles.linkWrapper}>
          <div ref={button}>
            <IconButton
              ariaLabel={t('collection.collectionCard.ariaLabelLink', {
                title,
              })}
              aria-hidden={true}
              icon={<IconArrowRight aria-hidden/>}
              onClick={moveToCollectionPage}
              size="default"
            />
          </div>
        </div>
      </div>
    </Link>
  );
};

export default CollectionCard;
