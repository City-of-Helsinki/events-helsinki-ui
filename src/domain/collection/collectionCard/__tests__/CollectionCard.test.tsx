import React from 'react';

import { ROUTES } from '../../../../domain/app/routes/constants';
import { CollectionFieldsFragment } from '../../../../generated/graphql';
import { fakeCollection } from '../../../../util/mockDataUtils';
import { render, screen, userEvent } from '../../../../util/testUtils';
import CollectionCard from '../CollectionCard';

const description = 'Lorem ipsum';
const heroImage = 'imageurl.png';
const slug = 'collection-title';
const title = 'Collection title';

const collection = fakeCollection({
  id: slug,
  description: { fi: description },
  slug,
  heroImage: { url: heroImage },
  title: { fi: title },
}) as CollectionFieldsFragment;

test('matches snapshot', () => {
  const { container } = render(
    <CollectionCard collection={collection} size="lg" />
  );

  expect(screen.getByText(title)).toBeInTheDocument();
  expect(screen.getByText(description)).toBeInTheDocument();
  expect(container.firstChild).toMatchSnapshot();
});

test('should hide desciprion', () => {
  render(<CollectionCard collection={collection} size="md" />);

  expect(screen.getByText(title)).toBeInTheDocument();
  expect(screen.queryByText(description)).not.toBeInTheDocument();
});

test('should go to collection page by clicking button', () => {
  const { history } = render(
    <CollectionCard collection={collection} size="lg" />
  );

  userEvent.click(screen.getByRole('button', { hidden: true }));
  expect(history.location.pathname).toBe(
    `/fi${ROUTES.COLLECTION.replace(':slug', slug)}`
  );
});

test('should go to collection page by clicking link', () => {
  const { history } = render(
    <CollectionCard collection={collection} size="lg" />
  );

  userEvent.click(screen.getByRole('link'));
  expect(history.location.pathname).toBe(
    `/fi${ROUTES.COLLECTION.replace(':slug', slug)}`
  );
});
