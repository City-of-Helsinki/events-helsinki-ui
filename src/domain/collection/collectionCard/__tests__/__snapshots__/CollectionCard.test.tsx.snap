// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  aria-label="Siirry kokoelmaan: Collection title"
  class="collectionCardWrapper lgSize"
  href="/fi/collection/collection-title"
>
  <div
    aria-hidden="true"
    class="imageWrapper"
    style="background-image: url(imageurl.png);"
  />
  <div
    class="textWrapper"
  >
    <div
      class="countWrapper"
    >
      <div
        class="count"
      >
        0 kpl
      </div>
    </div>
    <div
      class="titleWrapper"
    >
      <span
        class="title"
      >
        Collection title
      </span>
      <div
        class="description"
      >
        Lorem ipsum
      </div>
    </div>
    <div
      class="linkWrapper"
    >
      <div>
        <button
          aria-hidden="true"
          aria-label="Siirry kokoelmaan: Collection title"
          class="iconButton default backgroundDefault"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M12.586 7.05L14 5.636 20.364 12 14 18.364l-1.414-1.414 3.95-3.95H4v-2h12.535z"
                fill="currentColor"
              />
            </g>
          </svg>
        </button>
      </div>
    </div>
  </div>
</a>
`;
